locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -1608,7 +1606,35 @@ protected SshMachineLocation createTemporarySshMachineLocation(HostAndPort hostA
         }
     }
 
-    
+    /**
+     * Creates a temporary WinRM machine location (i.e. will not be persisted), which uses the given credentials.
```

Comment:
Keeping "various extra persisted location" files is definitely a bug.
Such extra files are called "dangling references" in Apache Brooklyn and we have validation about it called minRequiredHealthy which checks a healthy quorum check. If the minRequiredHealthy validation doesn't have healthy objects quorum, Brooklyn fails to start.
These failures are reported several times by clients.
We may want to keep some of the extra locations for history and durability reasons, but then we have to modify the minRequiredHealthy validation accordingly.